src/cplus_plugin/api/request.py

```diff
@@ -238,66 +766,152 @@ def start_upload_layer(self, file_path, component_type):
             "size": file_size,
             "number_of_parts": math.ceil(file_size / CHUNK_SIZE),
         }
-        response = self.post(self.urls.layer_upload_start(), payload)
-        result = response.json()
-        if response.status_code != 201:
+        result, status_code = self.post(self.urls.layer_upload_start(), payload)
+        if status_code != 201:
             raise CplusApiRequestError(result.get("detail", ""))
         return result
 
-    def finish_upload_layer(self, layer_uuid, upload_id, items):
+    def finish_upload_layer(
+        self,
+        layer_uuid: str,
+        upload_id: typing.Union[str, None],
+        items: typing.Union[typing.List[dict], None],
+    ) -> dict:
+        """Request for finishing layer upload
+
+        :param layer_uuid: UUID of the uploaded layer
+        :type layer_uuid: str
+
+        :param upload_id: Upload ID of the multipart upload, optional,
+            defaults to None
+        :type upload_id: str
+
+        :param items: List of uploaded items for multipart upload, optional,
+            defaults to None
+        :type items: typing.Union[typing.List[dict], None]
+
+        :return: Dictionary containing the UUID, name, size of the upload file
+        :rtype: dict
+        """
         payload = {}
         if upload_id:
             payload["multipart_upload_id"] = upload_id
         if items:
             payload["items"] = items
-        response = self.post(self.urls.layer_upload_finish(layer_uuid), payload)
-        result = response.json()
+        result, _ = self.post(self.urls.layer_upload_finish(layer_uuid), payload)
         return result
 
-    def abort_upload_layer(self, layer_uuid, upload_id):
+    def abort_upload_layer(self, layer_uuid: str, upload_id: str) -> bool:
+        """Aborting layer upload
+
+        :param layer_uuid: UUID of a Layer that is currently being uploaded
+        :type layer_uuid: str
+
+        :param upload_id: Multipart Upload ID
+        :type upload_id: str
+
+        :raises CplusApiRequestError: If the abort is failed
+
+        :return: True if upload is successfully aborted
+        :rtype: bool
+        """
         payload = {"multipart_upload_id": upload_id, "items": []}
-        response = self.post(self.urls.layer_upload_abort(layer_uuid), payload)
-        if response.status_code != 204:
-            result = response.json()
+        result, status_code = self.post(
+            self.urls.layer_upload_abort(layer_uuid), payload
+        )
+        if status_code != 204:
             raise CplusApiRequestError(result.get("detail", ""))
         return True
 
-    def submit_scenario_detail(self, scenario_detail):
-        log_response(scenario_detail, "scenario_detail")
-        response = self.post(self.urls.scenario_submit(), scenario_detail)
-        result = response.json()
-        if response.status_code != 201:
+    def submit_scenario_detail(self, scenario_detail: dict) -> bool:
```

Comment:
This is incorrect return type if the function returning string of scenario detail UUID